app/provider.tsx

```diff
@@ -130,7 +133,9 @@ export function Providers({ children }: { children: React.ReactNode }) {
           <NotificationProvider>
             <ThemeProvider theme={theme}>
               <PostHogProvider client={posthog}>
-                <QuestsContextProvider>{children}</QuestsContextProvider>
+                <WalletProvider>
```

Comment:
I don't think you need a walletProvider here to do the job, it adds unnecessary complexity. 

What I would do instead is to use starknet react `useAccount()` hook to change URL based on the address you can get from it, you can do that directly on the `[addressOrDomain]/page.tsx` file. 

Here is an example of what i would look like. 

```tsx
export default function Page({ params }: AddressOrDomainProps) {
  // ... existing state declarations ...
  const [connectedAddress, setConnectedAddress] = useState<string | undefined>(); // address set as connected
  const { address } = useAccount(); // current address connected
  
  // Add this useEffect to handle address changes and URL updates
  useEffect(() => {    
    // Skip if the address hasn't changed or isn't defined
    if (!address || address === connectedAddress) return;
    
    // Update connected address state
    setConnectedAddress(address);
    
    // If we're not on the current user's profile, don't redirect
    if (address && addressOrDomain !== address && addressOrDomain !== undefined) {
      return;
    }
    
    // Redirect to the new address profile
    if (address) {
      router.push(`/${address}`);
    }
  }, [address, addressOrDomain, connectedAddress, router]);

  // ... rest of the component ...
}
```

I might have done some mistake here so I'll let you test and adapt but that solution would be way easier than adding a provider (starknet-react is your provider here)